recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,682 @@
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+        if self.settings.os == "Android" and self.options.opengl == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+
+        if self.settings.os != "Windows" and self.options.opengl == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.with_fontconfig and not self.options.with_freetype:
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.settings.os == "Macos":
+            del self.settings.os.version
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
+
+        if not self.options.with_doubleconversion and str(self.settings.compiler.libcxx) != "libc++":
+            raise ConanInvalidConfiguration('Qt without libc++ needs qt:with_doubleconversion. '
+                                            'Either enable qt:with_doubleconversion or switch to libc++')
+
+        if tools.os_info.is_linux:
+            if self.options.qtwebengine:
+                self.options.with_fontconfig = True
+
+        def _enablemodule(mod):
+            if mod != 'qtbase':
+                setattr(self.options, mod, True)
+            for req in self._submodules[mod]["depends"]:
+                _enablemodule(req)
+
+        for module in self._submodules:
+            if module != 'qtbase' and getattr(self.options, module):
+                _enablemodule(module)
+
+    def requirements(self):
+        if self.options.openssl:
+            self.requires("openssl/1.1.1g")
+        if self.options.with_pcre2:
+            self.requires("pcre2/10.33")
+
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+        # if self.options.with_libiconv: # QTBUG-84708
+        #     self.requires("libiconv/1.16")# QTBUG-84708
+        if self.options.with_doubleconversion and not self.options.multiconfiguration:
+            self.requires("double-conversion/3.1.5")
+        if self.options.with_freetype and not self.options.multiconfiguration:
+            self.requires("freetype/2.10.1")
+        if self.options.with_fontconfig:
+            self.requires("fontconfig/2.13.91")
+        if self.options.with_icu:
+            self.requires("icu/64.2")
+        if self.options.with_harfbuzz and not self.options.multiconfiguration:
+            self.requires("harfbuzz/2.6.8")
+        if self.options.with_libjpeg and not self.options.multiconfiguration:
+            self.requires("libjpeg/9d")
+        if self.options.with_libpng and not self.options.multiconfiguration:
+            self.requires("libpng/1.6.37")
+        if self.options.with_sqlite3 and not self.options.multiconfiguration:
+            self.requires("sqlite3/3.32.2")
```

Comment:
I see that qt include a copy of sqlite3 https://github.com/qt/qtbase/blob/dev/src/3rdparty/sqlite/sqlite3.h#L126
* Is `self.requires("sqlite3/3.32.2")` here used to compile qt or it is only here to track the dependencies versions ? 
* Same for all other deps, as it looks like qt build all dependencies from source. https://github.com/qt/qtbase/tree/dev/src/3rdparty

Reply:
not for those which have a "system" option in qt build.

Reply:
but why not then using system deps if **multiconfiguration** is set as well ? recipe now uses two different versions of dependencies depends on the option multiconfiguration!

Reply:
because as far as I know, sqlite3 does not provide multi configuration package

Reply:
as far as I know deps does not have to be multi configuration if the consumer is. Take Openssl for example, it is not a multi configuration package and this condition is not applied to it. 
All these dependencies are statically linked so there wont be any runtime dependency.

Did I miss something here!

Reply:
the problem is that if you are building qt as multi-configuration, it will use a release-only or debug-only build of openssl, depending on the `build_type` setting value at the time you build qt. If you then consume this multi-configuration qt, with a different value for `build_type` setting, then there is a high probability of breakage, if openssl's library name is different depending on the `build_type` for example.

Reply:
ohh I see, that is problematic :( probably the best approach is to only package qt as single config and generate qt's cmake script using conan bu that is also a very challenging task I assume.